students/adam_strong/lesson09/mailroom_oo/donor_models.py

```diff
@@ -93,3 +64,34 @@ def write_letter(self):
         '''Writes an individual letter'''
         form_letter = letter.format(self.name, self.total)
         return form_letter
+
+
+class DonorCollections(object):
+    '''DonorCollections class used to store and manipultate the main dict'''
+
+    def __init__(self, *args):
+        self.donors = {d.name: d for d in args}
+
+    def add_donor(self, tyname, amt):
+        '''Adds a new donor or updates an existing donor, returns thank you letter'''
+        tyname = tyname.title()
+        if self.donors.get(tyname) is None:
+            self.donors[tyname] = Donor(tyname, [amt])
+        else:
+            self.donors[tyname].add_donation(amt)
+        ty_output = ty_message.format(tyname, amt)
+        return ty_output
+
+    def make_report(self):
+        '''Creates and returns the report on the data'''
+        lines = []
+        aggregate = ['\n', report_head, break_line]
+        for donor_ob in self.donors.values():
+            lines.append([donor_ob.format_donor(), donor_ob.total])
+        [aggregate.append(item[0]) for item in sorted(lines, key=lambda i: i[1], reverse=True)]
```

Comment:
a more correct way would be:
```
aggregate += [item[0] for item in sorted(...)]
```
in your code, it generates None values in a list that are not used (append() returns None)